src/banman.h

```diff
@@ -77,7 +77,7 @@ class BanMan
 
     // Removes all ban entries that include net_addr
     void UnbanAll(const CNetAddr& net_addr);
-    void GetBanned(banmap_t& banmap);
+    void GetBanned(banmap_t& banmap, std::optional<const CNetAddr> filterForIp = {});
```

Comment:
#20671 was merged, so I'm now using `std::optional` instead of `boost::optional`. 

`std::optional<T>` is considered ill-formed, where `is_reference_v<T>` and also if `!is_destructible_v<T>`. So we can't use `std::optional<const CNetAddr&>` and can't forward declare `CNetAddr` and `CSubNet` in this file as incomplete types are not destructible.

Reply:
I think better than adding this `std::optional<CNetAddr>` would be to just overload the `GetBanned()` function, since most of the logic in the function is now specific to the one case where you pass a filterForIp. Those feel like almost two different functional interfaces.

I think better yet would be to do the filtering logic inside the RPC `listbanned()` function. It's not required by any other callers, so I don't think there's a need to add it to banman. If other callers need the same filtering logic at some point in the future, the logic can be moved into banman at that point.

Reply:
That sounds good. Updated code accordingly.